### PR TITLE
ORCA-731 Force PR authors to rebase if master has diverged

### PR DESCRIPTION
**Pull request by pcalley**
Adds an "UnDiverged" apply requirement which forces users to rebase or merge if using the 'merge' checkout strategy and the local version of master on Atlantis has diverged from remote. This corresponds with the existing warning given to users during the plan phase.

### CODE REVIEW

File: runatlantis.io/docs/apply-requirements.md [hunk 1]
```diff
@@ -135,6 +135,41 @@ In Azure DevOps, all pull requests are mergeable unless there is a conflict. You
 At this time, the Azure DevOps client only supports merging using the default 'no fast-forward' strategy. Make sure your branch policies permit this type of merge.
 :::
 
+###UnDiverged
```
[the remark this answers is not in the document]
Comment (pcalley) [reply runatlantis.io/docs/apply-requirements.md]: Fixed these ones.

File: server/events/command_context.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ type CommandContext struct {
 	// set our own build statuses which can affect mergeability if users have
 	// required the Atlantis status to be successful prior to merging.
 	PullMergeable bool
+	PullDiverged  bool
```
[the remark this answers is not in the document]
Comment (pcalley) [reply server/events/command_context.go]: Yeah I added and never used it. Took it out.

File: server/events/yaml/valid/global_cfg.go [hunk 3]
```diff
@@ -119,18 +121,20 @@ func NewGlobalCfgWithHooks(allowRepoCfg bool, mergeableReq bool, approvedReq boo
 // approvedReq is true if users want to set the approved apply requirement
 // for all repos.
 // Deprecated: use NewGlobalCfgFromArgs
-func NewGlobalCfg(allowRepoCfg bool, mergeableReq bool, approvedReq bool) GlobalCfg {
+func NewGlobalCfg(allowRepoCfg bool, mergeableReq bool, approvedReq bool, unDivergedReq bool) GlobalCfg {
```
[the remark this answers is not in the document]
Comment (pcalley) [reply server/events/yaml/valid/global_cfg.go]: Yeah I didn't realize this was actually deprecated anyway and it's only used in tests. I reverted this change so we're not adding to deprecated functions and update the tests so they use GlobalCfgFromArgs and pass in a GlobalCfgArgs instead.

**Review by msarvar — APPROVED** [review]
lgtm

**Pull request merged**